### PR TITLE
Change `db_log_reader_thread()` and backtrace_thread()` to wait better

### PR DESCRIPTION
**Pull request by fruch**
## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I gave variables/functions meaningful self-explanatory names
- [x] I didn't leave commented-out/debugging code
- [x] I didn't copy-paste code
- [x] I added the relevant `backport` labels
- ~~[ ] New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- ~~[ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (`hydra unit-tests`)
- ~~[ ] I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -758,26 +758,22 @@ def backtrace_thread(self):
         """
         Keep reporting new coredumps found, every 30 seconds.
         """
-        while True:
-            if self.termination_event.isSet():
-                break
+        while not self.termination_event.isSet():
+            self.termination_event.wait(30)
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/cluster.py]: fixed

File: sdcm/cluster.py [hunk 2]
```diff
@@ -758,26 +758,22 @@ def backtrace_thread(self):
...
 
     def db_log_reader_thread(self):
         """
         Keep reporting new events from db log, every 30 seconds.
         """
-        while True:
-            if self.termination_event.isSet():
-                break
+        while not self.termination_event.isSet():
+            self.termination_event.wait(30)
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/cluster.py]: fixed

**Comment by amoskong** [comment]
@bentsi We need this for branch-2019.1

**Comment by bentsi** [comment]
backported to `branch-2019.1`

**Pull request merged**